framework/aster-frame/src/arch/x86/trap.rs

```diff
@@ -47,6 +48,40 @@ extern "sysv64" fn trap_handler(f: &mut TrapFrame) {
                 handle_virtual_exception(f, &ve_info);
             }
             &PAGE_FAULT => {
+                let page_fault_addr = unsafe { x86::controlregs::cr2() };
+
+                // Handle page fault from user space.
+                if (0..MAX_USERSPACE_VADDR).contains(&page_fault_addr) {
```

Comment:
0 is not sound to be included here, as that stand for the null pointer. But the framework can neglect this fact since `aster-nix` is fully responsible for not mapping from zero virtual page for the user (of course not to r/w within this virtual page). We can have an additional comment here waiving the responsibility restricting so. And let the framework user take care.